ignite/contrib/handlers/trains_logger.py

```diff
@@ -454,6 +454,8 @@ class TrainsLogger(BaseLogger):
 
     """
 
+    _bypass = None
```

Comment:
Can we clean this up somehow? I think class-level mutable attributes are asking for trouble. It _seems_ like it could be a constructor arg which would obviate the setters.

Reply:
IMO a constructor argument will only confuse users as this is a hack for testing and should not be used in any normal use-case...
I absolutely agree and share the sentiment 😄 , and I think we can simply forego the class-scope definition and stick to using `getattr(cls, "_bypass", None)` and `setattr(cls, "_bypass")` - it won't show in the interface or class definition, and will serve it's purpose, limited only to `bypass_mode()` and `set_bypass_mode()`.
What do you think?

Reply:
If it's strictly for testing, can we mock it?

Reply:
I think that mocking `TrainsLogger` instead of using the bypass will mean that relevant tests (those where we actually use the bypass, such as `test_integration()` and `test_integration_as_context_manager()`) will not actually test anything... The bypass mode itself is actually a kind of internal mock (only for the internal Task representation and server-side communication), and is currently used since Trains does not yet support an offline mode (which would, when available, be a good constructor argument).
Since I think the tests are important, I would suggest keeping the bypass mode (as suggested, not using a class-scope member), and replacing it with an offline mode, once we introduce that into Trains.
What do you think?

Reply:
@jkhenning OK, I see. How do you perform test integrations for other libs, do you mock like that ? Otherwise we can log to your demo server ?

Reply:
@vfdev-5 Yes, this is what we do in other integrations as well.
As for the demo server, we really try to keep it away from integration testing, mostly since we never want your integration testing dependent on a server that might be intermittently unavailable (due to load, maintenance, new version roll-outs etc.). Since the offline mode is on our to do list anyway, I really think this is the best solution going forward, and that the bypass is good enough for now 😄